src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3249 @@
+                ImportClasses(metaConfigExtensionFile, DefaultModuleInfoForMetaConfigResource, errors);
+
+                // Load all of the system resource schema files, searching
+                string resources;
+                foreach (var resourceRoot in allResourceRoots)
+                {
+                    resources = Path.Join(resourceRoot, "Schema");
+                    if (!Directory.Exists(resources))
+                    {
+                        continue;
+                    }
+
+                    foreach (var schemaFile in Directory.EnumerateFiles(resources, "*.schema.json", SearchOption.AllDirectories))
+                    {
+                        ImportClasses(schemaFile, s_defaultModuleInfoForResource, errors);
+                    }
+                }
+
+                // Load Regular and DSC PS modules
+                bool importInBoxResourcesImplicitly = false;
+                List<string> modulePaths = new List<string>();
+                if (modulePathList?.Count == 0)
+                {
+                    modulePaths.Add(Path.Join(configSystemPath, inboxModulePath));
+                    importInBoxResourcesImplicitly = true;
+                }
+                else
+                {
+                    if (modulePathList != null)
+                    {
+                        foreach (string moduleFolderPath in modulePathList)
+                        {
+                            if (!Directory.Exists(moduleFolderPath))
+                            {
+                                continue;
+                            }
+
+                            foreach (string moduleDir in Directory.EnumerateDirectories(moduleFolderPath))
+                            {
+                                modulePaths.Add(moduleDir);
+                            }
+                        }
+                    }
+                }
+
+                LoadDSCResourceIntoCache(errors, modulePaths, importInBoxResourcesImplicitly);
+            }
+        }
+
+        /// <summary>
+        /// Load DSC resources into Cache from moduleFolderPath.
+        /// </summary>
+        /// <param name="errors">Collection of any errors encountered during initialization.</param>
+        /// <param name="modulePathList">Module path from where DSC PS modules will be loaded.</param>
+        /// <param name="importInBoxResourcesImplicitly">
+        /// if module is inbox.
+        /// </param>
+        private static void LoadDSCResourceIntoCache(Collection<Exception> errors, List<string> modulePathList, bool importInBoxResourcesImplicitly)
+        {
+            foreach (string moduleDir in modulePathList)
+            {
+                if (!Directory.Exists(moduleDir)) continue;
+
+                var dscResourcesPath = Path.Join(moduleDir, "DscResources");
+                if (Directory.Exists(dscResourcesPath))
```

Comment:
Looks like these scenarios are not covered,
1) A DSC resource can have a versioning folder which contains "DscResource".
![image](https://user-images.githubusercontent.com/23668037/91911364-c6d63300-ec65-11ea-97a3-739c7811e14b.png)

2) class based DSC resource will not have "DscResource" folder.

Reply:
This is not the code that traverses PS Modules in search of resources - that is done in `function Get-DSCResourceModules` in `PSDesiredStateConfiguration.psm1`
It seems that versioned resources are processed fine:
![VersionBasedResourcs](https://user-images.githubusercontent.com/11860095/91919460-ee83c600-ec7a-11ea-90f5-b19043cbcca4.png)